cpp/src/gandiva/like_holder.cc

```diff
@@ -25,7 +25,7 @@ namespace gandiva {
 
 RE2 LikeHolder::starts_with_regex_(R"((\w|\s)*\.\*)");
 RE2 LikeHolder::ends_with_regex_(R"(\.\*(\w|\s)*)");
-RE2 LikeHolder::is_substr_regex_(R"(\.\*(\w|\s)*\.\*)");
+RE2 LikeHolder::is_substr_regex_(R"(\.\*(\w|\s|\W)*\.\*)");
```

Comment:
I am not sure if you can include all non-word characters.

From https://github.com/google/re2/wiki/Syntax:

\W | not word characters (≡ [^0-9A-Za-z_])
-- | --

Does RE2 support matching with non-ascii character sets?